Remotion/ObjectBinding/Web/UI/Controls/BocReferenceValueImplementation/BocReferenceValueBase.cs

```diff
@@ -964,9 +964,13 @@ string IBocReferenceValueBase.GetLabelText ()
       return GetLabelText();
     }
 
-    IEnumerable<string> IBocReferenceValueBase.GetValidationErrors ()
+    IEnumerable<PlainTextString> IBocReferenceValueBase.GetValidationErrors ()
     {
-      return GetRegisteredValidators().Where (v => !v.IsValid).Select (v => v.ErrorMessage).Distinct();
+      return GetRegisteredValidators()
+          .Where (v => !v.IsValid)
+          .Select (v => v.ErrorMessage)
+          .Select (PlainTextString.CreateFromText)
```

Comment:
Shouldn't we flip this and make the distinct first? 
(all occurances)

Also, should we take heart and add an extension method for `IEnumerable<PlaintextString> GetDistinctValidationErrors()(IEnumerable<IValidator>  this)`?

Reply:
Shouldn't make much of a difference in this case because both boil down to a pointer but it also should not hurt. If all the subclasses do the same thing why don't we introduce a method at the level of `GetRegisteredValidators`?

Reply:
There's a difference because PlainTextString.ToStirng() performs HTML Escaping of the string and then does a compare, unless I'm missing something?

Why extension methid instead of an alternative to GetRegisteredValidators()?
To me, it's easier to understand with a more modular approach. Of course, I could name the method `GetDistinctValidationErrorsForAllRegisteredValidators()` but this makes the usages of GetRegisteredValidators() (the impact in particular) more hidden and doesn't help with read-abilty. Also, there could be (in theory) other usecases for this API aside from processing the result of GetRegisteredValidators() when we put it in Remotion.Web.Utilities.ValidatorExtensions (or similar).

Reply:
`ToString` does do encoding, but `Distinct` should use as hash set and thus the `Equals` and `GetHashCode` methods, which do not encode. `PlainTextString.CreateFromText` just wraps into the string and also does not encode. So it should (fingers crossed) change anything (except that duplicates are not converted to structures first).

Ok I am going to change the `Distinct` order and introduce an extension method

Reply:
Correct, Distcint uses GetHascode and Euquals, so we don't operate on the encoded string. my bad. as you like.